src/contrib/cluster/Akka.Cluster.Tools/PublishSubscribe/DistributedPubSubMediator.cs

```diff
@@ -487,10 +487,23 @@ private void PutToRegistry(string key, IActorRef value)
                 _registry[_cluster.SelfAddress] = new Bucket(bucket.Owner, v, bucket.Content.SetItem(key, new ValueHolder(v, value)));
         }
 
-        private void IgnoreOrSendToDeadLetters(object message)
+        private void IgnoreOrSendToDeadLetters(IDistributedPubSubMessage message)
         {
             if (_settings.SendToDeadLettersWhenNoSubscribers)
-                Context.System.DeadLetters.Tell(new DeadLetter(message, Sender, Context.Self));
+            {
+                var topic = message switch
```

Comment:
Tries to extract the topic from the message so we can be as accurate as possible during the `DeadLetter` logging